### PR TITLE
Alerting: Consolidate contact points dropdown and add filter in alert rules

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**

Consolidates the contact point filters that we have across the app, and adds to the rules filters
Builds on the work from Sonia in https://github.com/grafana/grafana/pull/91245 🙌 
Some was pair/mob programmed together by alerting UI engineers

**Why do we need this feature?**

* Consolidation of code, so we fetch the contact points list in the same way everywhere (so it can all come from k8s API)
* Adding filtering to rules list so you can find simplified routing usages

**Who is this feature for?**

Alerting UI devs and users!

### CODE REVIEW

File: public/app/features/alerting/unified/components/notification-policies/EditNotificationPolicyForm.tsx [hunk 1]
```diff
@@ -168,14 +157,25 @@ export const AmRoutesExpandedForm = ({
...
-              className={formStyles.input}
-              onChange={(value) => onChange(mapSelectValueToString(value))}
-              options={receiversWithOnCallOnTop}
-              isClearable
+          render={({ field: { onChange, ref, value, ...field } }) => (
+            <ContactPointSelector
+              selectProps={{
+                ...field,
+                className: formStyles.input,
+                onChange: (value) => {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/notification-policies/EditNotificationPolicyForm.tsx:165]: It looks like we're no longer using `selectableValuesToStrings`, `selectableValueToString` and `mapSelectValueToString` here so we should be free to remove those.

It does look like the logic in this `onChange` handler is the exact same as in the default policy editor form, perhaps we can abstract this into a separate function?

Comment (tomratcliffe) [reply public/app/features/alerting/unified/components/notification-policies/EditNotificationPolicyForm.tsx:165]: We are using some of them actually :( 
`selectableValueToString` used in `selectableValuesToStrings`, which is used in `mapMultiSelectValueToStrings`, which is used in a few places 😞 

But it seems we can get rid of `mapSelectValueToString`

I'll move that other method into a helper util :+1:

**Review by gillesdemey — APPROVED** [review]
LGTM – no nits 🚀

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. This is a friendly reminder that the Steady [11.2.x Release Branch](https://github.com/grafana/grafana/tree/v11.2.x) was cut this week, and that it includes the changes committed up to https://github.com/grafana/grafana/commit/b4126d3bce3627e9460e2db0e870c8d984be8e84. This PR was merged after it, so we wanted to let you know that you can add the `backport v11.2.x` label to to backport it and make it part of the next 11.2.x Patch Release, just in case you may have missed the news.

Thank you!

**Pull request merged**